src/app/technical-record/technical-record.component.ts

```diff
@@ -1,116 +1,116 @@
+})
+export class TechnicalRecordComponent implements OnInit {
+  showAdrDetails: boolean;
+  adrDisplayParams: { [key: string]: boolean };
+  techRecord: FormGroup;
+
+  allOpened = false;
+  panels: { panel: string; isOpened: boolean }[] = [
+    { panel: 'panel1', isOpened: false },
+    { panel: 'panel2', isOpened: false },
+    { panel: 'panel3', isOpened: false },
+    { panel: 'panel4', isOpened: false },
+    { panel: 'panel5', isOpened: false },
+    { panel: 'panel6', isOpened: false },
+    { panel: 'panel7', isOpened: false },
+    { panel: 'panel8', isOpened: false },
+    { panel: 'panel9', isOpened: false },
+    { panel: 'panel10', isOpened: false }
+  ];
+
+  @Input() vehicleTechRecord: VehicleTechRecordModel;
+  @Input() activeRecord: TechRecord;
+  @Input() metaData: MetaData;
+  @Input() editState: VIEW_STATE;
+  @Input() testResultJson: TestResultModel[];
+  @Output() submitTechRecord = new EventEmitter<TechRecord>();
+  @Output() changeViewState = new EventEmitter<VIEW_STATE>();
+
+  constructor(
+    private dialog: MatDialog,
+    private allowedValues: TechnicalRecordValuesMapper,
+    public techRecHelpers: TechRecordHelpersService
+  ) {}
+
+  ngOnInit() {
+    this.techRecord = new FormGroup({});
+    this.adrDisplayParams = { showAdrDetails: !!this.activeRecord.adrDetails };
+  }
+
+  togglePanel() {
+    for (const panel of this.panels) {
+      panel.isOpened = !this.allOpened;
+    }
+    this.allOpened = !this.allOpened;
+  }
+
+  hasSecondaryVrms(vrms) {
+    return vrms && vrms.length > 1 && vrms.filter((vrm) => vrm.isPrimary === false).length > 0;
+  }
+
+  editTechRecord() {
+    this.changeViewState.emit(VIEW_STATE.EDIT);
+  }
+
+  cancelTechRecordEdit() {
+    this.changeViewState.emit(VIEW_STATE.VIEW_ONLY);
+    this.activeRecord = JSON.parse(JSON.stringify(this.activeRecord));
+    this.adrDisplayParams = { ...{ showAdrDetails: !!this.activeRecord.adrDetails } };
+
+    const controlNames = Object.keys(this.techRecord.controls);
+    for (let index = 0; index < controlNames.length; index++) {
+      this.techRecord.removeControl(controlNames[index]);
+    }
+    this.techRecord.updateValueAndValidity();
+  }
+
+  onSaveChanges({ valid, value }: { valid: boolean; value: any }) {
+    // if (valid) { TODO: Re-enable during ADR Validation ticket
+
+    const editedRecord: TechRecord = this.allowedValues.mapControlValuesToDataValues(
+      JSON.parse(JSON.stringify(value))
+    );
+    const mergedRecord: TechRecord = { ...this.activeRecord, ...editedRecord };
+
+    const dialogRef = this.dialog.open(AdrReasonModalComponent, {
+      width: '600px',
+      data: {
+        context: 'Enter reason for changing technical record'
+      }
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result && result.isSave) {
+        mergedRecord.reasonForCreation = result.data;
+        this.submitTechRecord.emit(mergedRecord);
+      }
+    });
+    // }
+  }
```

Comment:
Have we got unit tests for this file?

Reply:
This will be picked up as part of 10202.

Reply:
let's create a subtask for it please if we decide to do it later, the code coverage is good at the moment but we have a reminder and put a reference to it here please.
I think it would be best to have them pushed as commit code now but if it will take time or need more clarity we should reference it here and create a task.
Also bear in mind that tech-record will have different fields in fact, that will not be a 'copy paste' job.